sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -40,6 +40,22 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
 
   setupTestData()
 
+  test("SPARK-22431: table with nested type col with special char") {
```

Comment:
Move these two to `InMemoryCatalogedDDLSuite`

Reply:
Thanks @gatorsmile for your comments. I have addressed them in the latest commit.